workspaces/src/result.rs

```diff
@@ -149,6 +106,159 @@ impl CallExecutionDetails {
             .collect()
     }
 
+    /// Grab all logs from both the transaction and receipt outcomes.
+    pub fn logs(&self) -> Vec<&str> {
+        self.outcomes()
+            .iter()
+            .flat_map(|outcome| &outcome.logs)
+            .map(String::as_str)
+            .collect()
+    }
+}
+
+/// The result after evaluating the status of an execution. This can be [`ExecutionSuccess`]
+/// for successful executions or a [`ExecutionFailure`] for failed ones.
+#[derive(PartialEq, Eq, Clone)]
+#[non_exhaustive]
+pub struct ExecutionResult<T> {
+    /// Total gas burnt by the execution
+    pub total_gas_burnt: Gas,
+
+    /// Value returned from an execution. This is a base64 encoded str for a successful
+    /// execution or a `TxExecutionError` if a failed one.
+    pub(crate) value: T,
+    // pub(crate) transaction: ExecutionOutcome,
+    // pub(crate) receipts: Vec<ExecutionOutcome>,
+    pub(crate) details: ExecutionDetails,
+}
+
+impl<T: fmt::Debug> fmt::Debug for ExecutionResult<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("ExecutionResult")
+            .field("total_gas_burnt", &self.total_gas_burnt)
+            .field("transaction", &self.details.transaction)
+            .field("receipts", &self.details.receipts)
+            .field("value", &self.value)
+            .finish()
+    }
+}
+
+impl<T> Deref for ExecutionResult<T> {
+    type Target = ExecutionDetails;
+
+    fn deref(&self) -> &Self::Target {
+        &self.details
+    }
+}
```

Comment:
This feels like a strange impl to have, as it would confuse the API as these are very different types. Why not just expose the functions from `ExecutionResult` that you want from `ExecutionDetails`?